www/base/src/app/builders/builder/builder.controller.coffee

```diff
@@ -100,5 +100,9 @@ class Builder extends Controller
                 limit: $scope.numbuilds
                 order: '-number'
             $scope.buildrequests = builder.getBuildrequests(claimed:false)
+            $scope.buildrequests.onNew = (buildrequest) ->
```

Comment:
this will generate a large number of http requests

Reply:
Yes. Hopefully there are few unclaimed buildrequests for each builder. What would be the proper way to expose this? Create a new data API endpoint or adapt the getBuildrequests one to include the needed information?

Reply:
I would rather keep the buildbot API generic. The problem is actually not so much the rest API. build rest api already has a properties parameter to fetch more properties. We could add this to buildrequest as well.
The problem is for message queues messages which does not contain the information. So new buildrequests or new build events do not contain the properties required for the page.

A generic workaround would be in the data module. If there is additional query parameter, then the data module will automatically run the rest API for that single build/element in order to fetch the additional stuff. This would be a nice a generic way to handle the problem, while it generates more rest api calls, it simplifies the code on client side, by avoiding all those onNew specific calls.